src/main.ts

```diff
@@ -15,17 +15,28 @@ export const restClient = (apikey: string, restApiBase?: string, globalFetchOpti
   // Note: This does not include any custom interceptors or configurations
   const config = new Configuration({ apiKey: apikey });
   const SERVICE_BASE_URL = 'https://api.polygon.io'; // Fallback to default if not set
-  axios.interceptors.response.use(async (response) => {
+
+  // Create a new axios instance to avoid global interceptor pollution
+  const axiosInstance = axios.create();
+
+  axiosInstance.interceptors.response.use(async (response) => {
     if (globalFetchOptions?.pagination && response?.data?.next_url) {
       // If pagination is enabled and has a next_url, auto paginate the results and count
-      const nextResults: any = await axios.get(`${response.data.next_url}&apiKey=${apikey}`);
-      const { results, count } = nextResults;
-      return { ...response.data, results: [...results, ...response.data?.results], ...(response.data?.count && { count: response.data.count + count }) };
+      const nextResults: any = await axiosInstance.get(`${response.data.next_url}&apiKey=${apikey}`);
+      const { results, count } = nextResults.data;
+      return {
+        ...response,
+        data: {
+          ...response.data,
+          results: [...(response.data?.results || []), ...(results || [])],
+          ...(response.data?.count && count && { count: response.data.count + count })
```

Comment:
Our client should only return the data, that way it matches the old clients return type styling that was adjusted with the with this auto generated client.